components/yaml/introduction.rst

```diff
@@ -219,6 +216,58 @@ representation to the inline one:
         foo: bar
         bar: baz
 
+Indentation
+...........
+
+By default the YAML component will use 4 spaces for indentation. This can be
+changed using the second argument as follows::
+
+    // use 8 spaces for indentation
+    echo Yaml::dump($array, 2, 8);
+
+.. code-block:: yaml
+
+    foo: bar
+    bar:
+            foo: bar
+            bar: baz
+
+Invalid Types and Object Serialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default the YAML component will encode any "unsupported" type (i.e.
+resources and objects) as ``null``.
+
+Instead of encoding as ``null`` you can choose to throw an exception if an invalid
+type is encountered in either the dumper or parser as follows::
+
+    // throw an exception if a resource or object is encoutered
+    Yaml::dump($data, 2, 4, true);
+
+    // throw an exception if an encoded object is found in the YAML string
+    Yaml::parse($yaml, true);
+
+However, you can activate object support using the next argument::
+
+    $object = new \stdClass();
+    $object->hello = 'goodbye';
```

Comment:
Instead of `hello`, `goodbye`, etc. could we use `foo`, `bar`, etc.  as in the rest of the examples?